From c99f62ba1dc34c5111cff9ca3ce49b01bfd0774e Mon Sep 17 00:00:00 2001
From: Javier Martinez Canillas <javierm@redhat.com>
Date: Tue, 6 Feb 2018 11:16:28 +0100
Subject: [PATCH 204/206] Don't attempt to backtrace on grub_abort() for
 grub-emu

The emu platform doesn't have a grub_backtrace() implementation, so this
causes a build error. Don't attempt to call this when building grub-emu.

Signed-off-by: Javier Martinez Canillas <javierm@redhat.com>
---
 grub-core/kern/misc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/grub-core/kern/misc.c b/grub-core/kern/misc.c
index 04371ac49f2a..636f97e1ba12 100644
--- a/grub-core/kern/misc.c
+++ b/grub-core/kern/misc.c
@@ -1103,7 +1103,7 @@ static void __attribute__ ((noreturn))
 grub_abort (void)
 {
 #ifndef GRUB_UTIL
-#if defined(__i386__) || defined(__x86_64__)
+#if (defined(__i386__) || defined(__x86_64__)) && !defined(GRUB_MACHINE_EMU)
   grub_backtrace();
 #endif
 #endif
-- 
2.14.3

